File: src/TapeEquilibrium.java
public class TapeEquilibrium {

    public int solution(int[] A){
        int SumLeft=0;
        int SumRight = 0;
        int bestValue;
        for (int i = 0; i < A.length; i++)
        {
            SumRight += A[i];
        }

        bestValue=SumRight;
        for(int i=0;i<A.length;i++)
        {
            SumLeft += A[i];
            SumRight-=A[i];
            if (Math.abs(SumLeft - SumRight) < bestValue)
            {
                bestValue = Math.abs(SumLeft - SumRight);
            }


        }

        return bestValue;
    }

    public static void main(String args[]) {
        TapeEquilibrium TE = new TapeEquilibrium();
        System.out.println(TE.solution(new int[]{3,1,2,4,3}));
    }
}

File: src/JS/TapeEquilibrium.js
function TapeEquilibrium(A) {
  let totalSum = 0;
  for (let i = 0; i < A.length; i++) {
    totalSum += A[i];
  }
  //   let totalSum = A.reduce((acc, val) => acc + val, 0);
  let leftSum = A[0];
  let rightSum = totalSum - leftSum;
  let minDiff = Math.abs(leftSum - rightSum);
  for (let i = 1; i < A.length - 1; i++) {
    leftSum += A[i];
    rightSum -= A[i];

    const diff = Math.abs(leftSum - rightSum);
    if (diff < minDiff) {
      minDiff = diff;
    }
  }
  console.log(minDiff, "minDiff");
  console.log(totalSum, "totalsum");
  console.log(leftSum, "leftsum");
  console.log(rightSum, "rightsum");
  return minDiff;
}

TapeEquilibrium([3, 1, 2, 4, 3]);
